# 基金

In [2]:
import os, sys
import numpy as np
import pandas as pd

In [3]:
import akshare as ak
import baostock as bt
import tushare as ts

In [92]:
from ipywidgets import *
import ipywidgets as widgets
from matplotlib.pyplot import *

## 基金基本信息
  描述: 获取东方财富网站-天天基金网-基金数据-所有基金的基本信息数据  
  限量: 单次返回当前时刻所有历史数据    
  输入参数  
    名称	类型	必选	描述  
    -	-	-	-  
  输出参数  
    名称       类型    默认显示    描述  
    基金代码    str    Y          -  
    拼音缩写    str    Y          -  
    基金简称    str    Y          -  
    基金类型    str    Y          -  
    拼音全称    str    Y          -  

In [4]:
fund_em_fund_name_df = ak.fund_em_fund_name()
print(fund_em_fund_name_df)

         基金代码             拼音缩写             基金简称  基金类型  \
0      000001           HXCZHH           华夏成长混合   混合型   
1      000002           HXCZHH       华夏成长混合(后端)   混合型   
2      000003         ZHKZZZQA         中海可转债债券A   债券型   
3      000004         ZHKZZZQC         中海可转债债券C   债券型   
4      000005       JSZQXYDQZQ       嘉实增强信用定期债券  定开债券   
...       ...              ...              ...   ...   
12233  970018  FZZQJGWLGYCYZQA  方正证券金港湾六个月持有债券A   债券型   
12234  970019  FZZQJGWLGYCYZQC  方正证券金港湾六个月持有债券C   债券型   
12235  970020          XDJZJXA          信达价值精选A   混合型   
12236  970021          XDJZJXB          信达价值精选B   混合型   
12237  980003      TPYLGYGDCYZ      太平洋六个月滚动持有债   债券型   

                                                    拼音全称  
0                                  HUAXIACHENGZHANGHUNHE  
1                                  HUAXIACHENGZHANGHUNHE  
2                           ZHONGHAIKEZHUANZHAIZHAIQUANA  
3                           ZHONGHAIKEZHUANZHAIZHAIQUANC  
4                   

## 基金净值
### 开放式基金-实时数据

**[描述]**: 获取东方财富网站-天天基金网-基金数据, 此接口在每个交易日 16:00-23:00 更新当日的最新开放式基金净值数据

**[限量]**: 单次返回当前时刻所有历史数据

输入参数

名称	类型	必选	描述
-	-	-	-

输出参数  
名称	类型	默认显示	描述  
基金代码	str	Y	-  
基金简称	str	Y	-  
单位净值	float	Y	随时间变动  
累计净值	float	Y	随时间变动  
前交易日-单位净值	float	Y	随时间变动  
前交易日-累计净值	float	Y	随时间变动  
日增长值	float	Y	-  
日增长率	float	Y	-  
申购状态	str	Y	-  
赎回状态	str	Y	-  
手续费	str	Y	注意单位: %  

**[基金净值]**: 是又称基金单位净值，是每份基金单位的净资产价值，等于基金的总资产减去总负债后的余额再除以基金全部发行的单位份额总数。开放式基金的申购和赎回都以这个价格进行。

计算公式为：基金份额资产净值=（总资产-总负债）/基金份额总数。

其中，总资产是指基金拥有的所有资产，包括股票、债券、银行存款和其他有价证券等；

总负债是指基金运作及融资时所形成的负债，包括应付给他人的各项费用、应付资金利息等；

基金份额总数是指当时发行在外的基金份额的总量。

开放式基金的份额总数每天都在变化，因此须以当日交易结束后的统计数为准。在每个营业日收市后，将当日基金资产净值除以当日交易截止时的基金份额总数，就得出当日的份额资产净值。

In [5]:
fund_em_open_fund_daily_df = ak.fund_em_open_fund_daily()
print(fund_em_open_fund_daily_df)

         基金代码            基金简称 2021-03-17-单位净值 2021-03-17-累计净值 2021-03-16-单位净值  \
0      000828          泰达转型机遇          1.9890          2.2090          1.9080   
1      010135   泰达高研发创新6个月混合A          0.7817          0.7817          0.7529   
2      010136   泰达高研发创新6个月混合C          0.7806          0.7806          0.7518   
3      002190         农银新能源主题          2.4513          2.4513          2.3612   
4      162201        泰达宏利成长混合          1.4318          3.8283          1.3829   
...       ...             ...             ...             ...             ...   
10566  011609  易方达科创板50ETF联接C                                                   
10567  011094          诺德安盛纯债                                                   
10568  011683       华夏鼎华一年定开债                                                   
10569  009679    浙商惠隆39个月定开债券                                                   
10570  010699       东方红创新趋势混合                                                   

      2021-03-16-累计净值    日增

### 开放式基金-历史数据
接口: fund_em_open_fund_info

描述: 获取东方财富网站-天天基金网-基金数据-具体基金信息

限量: 单次返回当前时刻所有历史数据

输入参数  
名称	类型	必选	描述  
fund	str	Y	fund="710001"; 需要基金代码, 可以通过调用 fund_em_open_fund_daily 获取  
indicator	str	Y	indicator="单位净值走势"; 参见 fund_em_open_fund_info 参数一览表  
fund_em_open_fund_info 参数一览表  

参数名称	备注  
单位净值走势	-  
累计净值走势	-  
累计收益率走势	-  
同类排名走势	-  
同类排名百分比	-  
分红送配详情	-  
拆分详情	-  

输出参数-单位净值走势  
名称	类型	默认显示	描述  
净值日期	str	Y	-  
单位净值	str	Y	-  
日增长率	str	Y	注意单位: %  

In [6]:
fund_em_info_df = ak.fund_em_open_fund_info(fund="710001", indicator="单位净值走势")
print(fund_em_info_df)

            净值日期    单位净值  日增长率
0     2011-09-21  1.0000     0
1     2011-09-23  1.0000     0
2     2011-09-30  1.0001  0.01
3     2011-10-14  1.0005  0.04
4     2011-10-21  1.0007  0.02
...          ...     ...   ...
2293  2021-03-11  3.0888  1.97
2294  2021-03-12  3.0646 -0.78
2295  2021-03-15  2.9846 -2.61
2296  2021-03-16  2.9729 -0.39
2297  2021-03-17  3.0187  1.54

[2298 rows x 3 columns]


In [7]:
fund_em_info_df = ak.fund_em_open_fund_info(fund="710001", indicator="累计净值走势")
print(fund_em_info_df)

            净值日期    累计净值
0     2011-09-21  1.0000
1     2011-09-23  1.0000
2     2011-09-30  1.0001
3     2011-10-14  1.0005
4     2011-10-21  1.0007
...          ...     ...
2293  2021-03-11  3.0888
2294  2021-03-12  3.0646
2295  2021-03-15  2.9846
2296  2021-03-16  2.9729
2297  2021-03-17  3.0187

[2298 rows x 2 columns]


输出参数-累计收益率走势  
名称	类型	默认显示	描述  
净值日期	str	Y	-  
累计收益率	str	Y	注意单位: %  

In [8]:
fund_em_info_df = ak.fund_em_open_fund_info(fund="710001", indicator="累计收益率走势")
print(fund_em_info_df)

           净值日期  累计收益率
0    2020-09-17   0.00
1    2020-09-18   1.97
2    2020-09-21   1.51
3    2020-09-22   0.92
4    2020-09-23   1.30
..          ...    ...
113  2021-03-11  11.13
114  2021-03-12  10.26
115  2021-03-15   7.38
116  2021-03-16   6.96
117  2021-03-17   8.61

[118 rows x 2 columns]


输出参数-同类排名走势  
名称	类型	默认显示	描述  
报告日期	str	Y	-  
同类型排名-每日近三月排名	str	Y	-  
总排名-每日近三月排名	str	Y	-  

In [9]:
fund_em_info_df = ak.fund_em_open_fund_info(fund="710001", indicator="同类排名走势")
print(fund_em_info_df)

            报告日期  同类型排名-每日近三月排名 总排名-每日近三月排名
0     2013-01-04             53         533
1     2013-01-07             68         533
2     2013-01-08            102         533
3     2013-01-09            117         533
4     2013-01-10            129         533
...          ...            ...         ...
1989  2021-03-11            473        4312
1990  2021-03-12            604        4467
1991  2021-03-15            563        4311
1992  2021-03-16            727        4189
1993  2021-03-17            583        4167

[1994 rows x 3 columns]


输出参数-同类排名百分比  
名称	类型	默认显示	描述  
报告日期	str	Y	-  
同类型排名-每日近3月收益排名百分比	str	Y	-  

In [10]:
fund_em_info_df = ak.fund_em_open_fund_info(fund="710001", indicator="同类排名百分比")
print(fund_em_info_df)

            报告日期  同类型排名-每日近3月收益排名百分比
0     2013-01-04               90.06
1     2013-01-07               87.24
2     2013-01-08               80.86
3     2013-01-09               78.05
4     2013-01-10               75.80
...          ...                 ...
1989  2021-03-11               89.03
1990  2021-03-12               86.48
1991  2021-03-15               86.94
1992  2021-03-16               82.65
1993  2021-03-17               86.01

[1994 rows x 2 columns]


输出参数-分红送配详情  
名称	类型	默认显示	描述  
年份	str	Y	-  
权益登记日	str	Y	-  
除息日	str	Y	-  
每份分红	str	Y	-  
分红发放日	str	Y	-  

In [11]:
fund_em_info_df = ak.fund_em_open_fund_info(fund="161606", indicator="分红送配详情")
print(fund_em_info_df)

      年份       权益登记日         除息日          每份分红       分红发放日
0  2021年  2021-01-19  2021-01-19  每份派现金0.0050元  2021-01-21
1  2007年  2007-10-25  2007-10-25  每份派现金0.7700元  2007-10-29
2  2007年  2007-03-12  2007-03-12  每份派现金1.1100元  2007-03-14
3  2006年  2006-05-30  2006-05-30  每份派现金0.0600元  2006-06-01


输出参数-拆分详情  
名称	类型	默认显示	描述  
年份	str	Y	-  
拆分折算日	str	Y	-  
拆分类型	str	Y	-  
拆分折算比例	str	Y	-  

In [12]:
fund_em_info_df = ak.fund_em_open_fund_info(fund="161606", indicator="拆分详情")
print(fund_em_info_df)

        年份    拆分折算日     拆分类型   拆分折算比例 Unnamed: 4
0  暂无拆分信息!  暂无拆分信息!  暂无拆分信息!  暂无拆分信息!    暂无拆分信息!


### 货币型基金-实时数据
接口: fund_em_money_fund_daily  
描述: 获取东方财富网站-天天基金网-基金数据-货币型基金收益, 此接口数据每个交易日 16:00～23:00  
限量: 单次返回当前时刻所有历史数据  
输入参数  
名称	类型	必选	描述  
-	-	-	-  
输出参数  
名称	类型	默认显示	描述  
基金代码	str	Y	-  
基金简称	str	Y	-  
当前交易日-万份收益	float	Y	-  
当前交易日-7日年化%	float	Y	-  
当前交易日-单位净值	float	Y	-  
前一交易日-万份收益	float	Y	-  
前一交易日-7日年化%	float	Y	-  
前一交易日-单位净值	float	Y	-  
日涨幅	str	Y	-  
成立日期	str	Y	-  
基金经理	str	Y	-  
手续费	str	Y	-  
可购全部	str	Y	-  

In [13]:
fund_em_money_fund_daily_df = ak.fund_em_money_fund_daily()
print(fund_em_money_fund_daily_df)

       基金代码         基金简称 2021-03-17-万份收益 2021-03-17-7日年化% 2021-03-17-单位净值  \
2    690210  民生加银现金增利货币B          0.5143          4.7750%             ---   
3    690010  民生加银现金增利货币A          0.4578          4.5280%             ---   
4    005151    红土创新优淳货币B          1.2221          3.3690%             ---   
5    091005        大成货币B          0.5432          3.3330%             ---   
6    002733      上银慧盈利货币          0.7991          3.2790%             ---   
..      ...          ...             ...              ...             ...   
468  003525     兴银现金收益货币          0.3217          1.0570%             ---   
469  000439       国金鑫盈货币          0.3101          1.0400%             ---   
470  630012    华商现金增利货币A          0.2619          0.9380%             ---   
471  420106    天弘现金管家货币B             ---              ---             ---   
472  020032    国泰现金管理货币B             ---              ---             ---   

    2021-03-16-万份收益 2021-03-16-7日年化% 2021-03-16-单位净值  日涨幅        成立日期   基金经

### 货币型基金-历史数据  
接口: fund_em_money_fund_info  
描述: 获取东方财富网站-天天基金网-基金数据-货币型基金历史净值  
限量: 单次返回当前时刻所有历史数据  
输入参数  
名称	类型	必选	描述  
fund	str	Y	fund="000009"; 需要基金代码, 可以通过调用 fund_em_money_fund_daily 获取  
输出参数  
名称	类型	默认显示	描述  
净值日期	str	Y	-  
每万份收益	str	Y	-  
7日年化收益率	float	Y	-  
申购状态	str	Y	-  
赎回状态	str	Y	-  

In [14]:
fund_em_money_fund_info_df = ak.fund_em_money_fund_info(fund="000009")
print(fund_em_money_fund_info_df)

            净值日期    每万份收益 7日年化收益率  申购状态  赎回状态
0     2021-03-17   0.5847  2.2610  开放申购  开放赎回
1     2021-03-16   0.5841  2.2850  开放申购  开放赎回
2     2021-03-15   0.5811  2.2930  开放申购  开放赎回
3     2021-03-14   0.6362  2.3220  开放申购  开放赎回
4     2021-03-13   0.6363  2.3240  开放申购  开放赎回
...          ...      ...     ...   ...   ...
2740  2013-03-29   0.6744  2.8920  开放申购  开放赎回
2741  2013-03-28   0.7068  3.0420  开放申购  开放赎回
2742  2013-03-27   0.3729  3.1750  开放申购  开放赎回
2743  2013-03-26   0.9561  3.4820  开放申购  开放赎回
2744  2013-03-25  21.2775  3.4850   封闭期   封闭期

[2745 rows x 5 columns]


### 理财型基金-实时数据  
接口: fund_em_financial_fund_daily  
描述: 获取东方财富网站-天天基金网-基金数据-理财型基金-实时数据, 此接口数据每个交易日 16:00～23:00 更新  
限量: 单次返回当前时刻所有历史数据  
输入参数  
名称	类型	必选	描述  
-	-	-	-  
输出参数  
名称	类型	默认显示	描述  
序号	int	Y	-  
基金代码	str	Y	-  
基金简称	str	Y	-  
上一期年化收益率	float	Y	-  
当前交易日-万份收益	float	Y	-  
当前交易日-7日年华	float	Y	-  
前一个交易日-万份收益	float	Y	-  
前一个交易日-7日年华	float	Y	-  
封闭期	float	Y	-  
申购状态	str	Y	-  

In [15]:
fund_em_financial_fund_daily_df = ak.fund_em_financial_fund_daily()
print(fund_em_financial_fund_daily_df)

None


### 理财型基金-历史数据  
接口: fund_em_financial_fund_info  
描述: 获取东方财富网站-天天基金网-基金数据-理财型基金-基金净值数据  
限量: 单次返回当前时刻所有历史数据  
输入参数  
名称	类型	必选	描述  
fund	str	Y	fund="000134"; 需要基金代码, 可以通过调用 fund_em_financial_fund_daily 获取  
输出参数  
名称	类型	默认显示	描述  
净值日期	str	Y	-  
每万份收益	str	Y	-  
7日年化收益率	float	Y	-  
申购状态	str	Y	-  
赎回状态	str	Y	-  

In [16]:
fund_em_financial_fund_info_df = ak.fund_em_financial_fund_info(fund="000134")
print(fund_em_financial_fund_info_df)

           净值日期   每万份收益 7日年化收益率  申购状态  赎回状态
0    2021-03-17  1.0063  1.0180  开放申购  开放赎回
1    2021-03-16  1.0060  1.0177  开放申购  开放赎回
2    2021-03-15  1.0059  1.0176  开放申购  开放赎回
3    2021-03-12  1.0057  1.0174  开放申购  开放赎回
4    2021-03-11  1.0056  1.0173  开放申购  开放赎回
..          ...     ...     ...   ...   ...
173  2020-07-01  0.9998  0.9998  暂停申购  开放赎回
174  2020-06-30  0.9995  0.9995  暂停申购  开放赎回
175  2020-06-29  0.9994  0.9994  暂停申购  开放赎回
176  2020-06-24  0.9987  0.9987  暂停申购  开放赎回
177  2020-06-23  0.9982  0.9982  暂停申购  开放赎回

[178 rows x 5 columns]


### 分级基金-实时数据  
接口: fund_em_graded_fund_daily  
描述: 获取东方财富网站-天天基金网-基金数据-分级基金-实时数据, 此接口数据每个交易日 16:00～23:00  
限量: 单次返回当前时刻所有历史数据  
输入参数  
名称	类型	必选	描述  
-	-	-	-  
输出参数  
名称	类型	默认显示	描述  
基金代码	str	Y	-  
基金简称	str	Y	-  
单位净值	float	Y	-  
累计净值	float	Y	-  
前交易日-单位净值	float	Y	-  
前交易日-累计净值	float	Y	-  
日增长值	float	Y	-  
日增长率	float	Y	注意单位  
市价	str	Y	-  
折价率	str	Y	-  
手续费	str	Y	-  

In [17]:
fund_em_graded_fund_daily_df = ak.fund_em_graded_fund_daily()
print(fund_em_graded_fund_daily_df)

     基金代码           基金简称 2021-03-17-单位净值 2021-03-17-累计净值 2021-03-16--单位净值  \
0  150325  工银中证高铁产业指数分级A                                                    
1  150326  工银中证高铁产业指数分级B                                                    
2  164820   工银中证高铁产业指数分级                                                    

  2021-03-16--累计净值 日增长值 日增长率 市价 折价率    手续费  
0                                           
1                                           
2                                    1.00%  


### 分级基金-历史数据  
接口: fund_em_graded_fund_info  
描述: 获取东方财富网站-天天基金网-基金数据-分级基金-历史数据  
限量: 单次返回当前时刻所有历史数据  
输入参数  
名称	类型	必选	描述  
fund	str	Y	fund="150232"; 需要基金代码, 可以通过调用 fund_em_graded_fund_daily 获取  
输出参数  
名称	类型	默认显示	描述  
净值日期	str	Y	-  
单位净值	float	Y	-  
累计净值	float	Y	-  
日增长率	float	Y	注意单位: %; 日增长率为空原因如下: 1. 非交易日净值不参与日增长率计算(灰色数据行). 2. 上一交易日净值未披露, 日增长率无法计算.  
申购状态	str	Y	-  
赎回状态	str	Y	-  

In [18]:
fund_em_graded_fund_info_df = ak.fund_em_graded_fund_info(fund="150232")
print(fund_em_graded_fund_info_df)

            净值日期    单位净值    累计净值   日增长率 申购状态 赎回状态
0     2020-11-09  1.1120           6.38  封闭期  封闭期
1     2020-11-06  1.0453           2.13  封闭期  封闭期
2     2020-11-05  1.0235           4.12  封闭期  封闭期
3     2020-11-04  0.9830           0.83  封闭期  封闭期
4     2020-11-03  0.9749           1.72  封闭期  封闭期
...          ...     ...     ...    ...  ...  ...
1338  2015-05-20  0.9988  0.9988  -0.03  封闭期  封闭期
1339  2015-05-19  0.9991  0.9991  -0.02  封闭期  封闭期
1340  2015-05-18  0.9993  0.9993  -0.04  封闭期  封闭期
1341  2015-05-15  0.9997  0.9997  -0.03  封闭期  封闭期
1342  2015-05-14  1.0000  1.0000         封闭期  封闭期

[1343 rows x 6 columns]


### 场内交易基金-实时数据  
接口: fund_em_etf_fund_daily  
描述: 获取东方财富网站-天天基金网-基金数据-场内交易基金-实时数据, 此接口数据每个交易日 16:00～23:00  
限量: 单次返回当前时刻所有数据  
输入参数  
名称	类型	必选	描述  
-	-	-	-  
输出参数  
名称	类型	默认显示	描述  
基金代码	str	Y	-  
基金简称	str	Y	-  
类型	float	Y	-  
当前交易日-单位净值	float	Y	会返回具体的日期值作为字段  
当前交易日-累计净值	float	Y	会返回具体的日期值作为字段  
前一个交易日-单位净值	float	Y	会返回具体的日期值作为字段  
前一个交易日-累计净值	float	Y	会返回具体的日期值作为字段  
增长值	float	Y	-  
增长率	str	Y	-  
市价	str	Y	-  
折价率	str	Y	-  

In [19]:
fund_em_etf_fund_daily_df = ak.fund_em_etf_fund_daily()
print(fund_em_etf_fund_daily_df)

       基金代码             基金简称      类型 2021-03-17-单位净值 2021-03-17-累计净值  \
0    511010   国泰上证5年期国债ETF行情  ETF-场内        121.8370          1.2310   
1    159926    嘉实中证中期国债ETF行情  ETF-场内        114.1560          1.1420   
2    511280   华夏3-5年信用债ETF行情  ETF-场内        111.7911          1.1179   
3    511310  富国中证10年期国债ETF行情  ETF-场内        111.0944          1.1109   
4    511260  国泰上证10年期国债ETF行情  ETF-场内        110.7240          1.1070   
..      ...              ...     ...             ...             ...   
403  159880  鹏华国证有色金属行业ETF行情  ETF-场内             ---             ---   
404  516300  华泰柏瑞中证1000ETF行情  ETF-场内             ---             ---   
405  159876      华宝有色金属ETF行情  ETF-场内             ---             ---   
406  159858   南方中证创新药产业ETF行情  ETF-场内             ---             ---   
407  159871    银华中证有色金属ETF行情  ETF-场内             ---             ---   

    2021-03-16-单位净值 2021-03-16-累计净值     增长值    增长率        市价    折价率  
0          121.7930          1.2310  0.0440  0.04%  121.6180  0.1

### 场内交易基金-历史数据  
接口: fund_em_etf_fund_info  
描述: 获取东方财富网站-天天基金网-基金数据-场内交易基金-历史净值数据  
限量: 单次返回当前时刻所有历史数据  
输入参数  
名称	类型	必选	描述  
fund	str	Y	fund="511280"; 需要基金代码, 可以通过调用 fund_em_etf_fund_daily 获取  
输出参数  
名称	类型	默认显示	描述  
净值日期	str	Y	-  
单位净值	float	Y	-  
累计净值	float	Y	-  
日增长率	float	Y	注意单位(%)  
申购状态	str	Y	-  
赎回状态	str	Y	-  

In [20]:
fund_em_etf_fund_info_df = ak.fund_em_etf_fund_info(fund="511280")
print(fund_em_etf_fund_info_df)

           净值日期      单位净值    累计净值  日增长率  申购状态  赎回状态
0    2021-03-17  111.7911  1.1179  0.07  场内买入  场内卖出
1    2021-03-16  111.7177  1.1172  0.00  场内买入  场内卖出
2    2021-03-15  111.7170  1.1172  0.04  场内买入  场内卖出
3    2021-03-12  111.6774  1.1168  0.01  场内买入  场内卖出
4    2021-03-11  111.6626  1.1166  0.04  场内买入  场内卖出
..          ...       ...     ...   ...   ...   ...
685  2018-05-18  100.1054  1.0011        场内买入  场内卖出
686  2018-05-14  100.0804  1.0008  0.02  场内买入  场内卖出
687  2018-05-11    1.0006  1.0006        场内买入  场内卖出
688  2018-05-04    1.0003  1.0003  0.03  场内买入  场内卖出
689  2018-05-03    1.0000  1.0000        场内买入  场内卖出

[690 rows x 6 columns]


### 香港基金-历史数据  
接口: fund_em_hk_fund_hist  
描述: 获取东方财富网站-天天基金网-基金数据-香港基金-历史净值明细  
限量: 单次返回指定 code 和 symbol 所有历史数据  
输入参数  
名称	类型	必选	描述  
code	str	Y	code="1002200683"; 需要香港基金代码, 可以通过调用 fund_em_hk_rank 获取  
symbol	str	Y	symbol="历史净值明细"; choice of {"历史净值明细", "分红送配详情"}  
输出参数-历史净值明细  
名称	类型	默认显示	描述  
净值日期	str	Y	-  
单位净值	float	Y	-  
日增长值	float	Y	-  
日增长率	float	Y	注意单位: %  
单位	str	Y	-  

In [21]:
fund_em_hk_fund_hist_df = ak.fund_em_hk_fund_hist(code='1002200683', symbol="历史净值明细")
print(fund_em_hk_fund_hist_df)

           净值日期     单位净值    日增长值    日增长率 单位
0    2021-03-17   9.8975  0.0267  0.2705  元
1    2021-03-16   9.8708  0.0004  0.0041  元
2    2021-03-15   9.8704 -0.0046 -0.0466  元
3    2021-03-12   9.8750  0.0553  0.5632  元
4    2021-03-11   9.8197 -0.0092 -0.0936  元
..          ...      ...     ...     ... ..
201  2020-05-14  10.0774  0.0214  0.2128  元
202  2020-05-13  10.0560   0.021  0.2093  元
203  2020-05-12  10.0350   0.035    0.35  元
204  2020-05-11  10.0000       0       0  元
205  2020-05-08  10.0000                  元

[206 rows x 5 columns]


输出参数-分红送配详情  
名称	类型	默认显示	描述  
年份	str	Y	-  
权益登记日	str	Y	-  
除息日	str	Y	-  
分红发放日	str	Y	-  
分红金额	float	Y	-  
单位	str	Y	-  

In [22]:
fund_em_hk_fund_hist_df = ak.fund_em_hk_fund_hist(code='1002200683', symbol="分红送配详情")
print(fund_em_hk_fund_hist_df)

     年份       权益登记日         除息日       分红发放日    分红金额 单位
0  2021  2021-02-26  2021-02-26  2021-03-12  0.0489  元
1  2021  2021-01-29  2021-01-29  2021-02-19  0.0478  元
2  2020  2020-12-31  2020-12-31  2021-01-15  0.0522  元
3  2020  2020-11-30  2020-11-30  2020-12-14  0.0552  元
4  2020  2020-10-30  2020-10-30  2020-11-13  0.0573  元
5  2020  2020-09-30  2020-09-30  2020-10-16  0.0527  元
6  2020  2020-08-31  2020-08-31  2020-09-14  0.0545  元
7  2020  2020-07-31  2020-07-31  2020-08-14  0.0593  元
8  2020  2020-06-30  2020-06-30  2020-07-15  0.0669  元


## 基金排行

### 开放式基金排行
接口: fund_em_open_fund_rank  
描述: 获取东方财富网-数据中心-开放式基金排行  
限量: 单次返回当前时刻所有数据  
输入参数  
名称	类型	必选	描述  
symbol	str	Y	symbol="全部"; choice of {"全部", "股票型", "混合型", "债券型", "指数型", "QDII", "LOF", "FOF"}  
输出参数  
名称	类型	默认显示	描述  
序号	int	Y	-  
基金代码	str	Y	-  
基金简称	str	Y	-  
日期	str	Y	-  
单位净值	float	Y	-  
累计净值	float	Y	-  
日增长率	float	Y	注意单位: %  
近1周	float	Y	注意单位: %  
近1月	float	Y	注意单位: %  
近3月	float	Y	注意单位: %  
近6月	float	Y	注意单位: %  
近1年	float	Y	注意单位: %  
近2年	float	Y	注意单位: %  
近3年	float	Y	注意单位: %  
今年来	float	Y	注意单位: %  
成立来	float	Y	注意单位: %  
自定义	float	Y	注意单位: %  
手续费	str	Y	-  

In [23]:
fund_em_open_fund_rank_df = ak.fund_em_open_fund_rank(symbol="全部")
print(fund_em_open_fund_rank_df)

        序号    基金代码            基金简称          日期    单位净值    累计净值   日增长率    近1周  \
0        1  005297         南华丰淳混合C  2021-03-17  1.9092  1.9092   1.50  32.43   
1        2  005296         南华丰淳混合A  2021-03-17  1.9728  1.9728   1.51  32.39   
2        3  460010    华泰柏瑞亚洲领导企业混合  2021-03-16  1.7050  1.7050   2.03   8.19   
3        4  164908   交银中证环境治理(LOF)  2021-03-17  0.5280  0.5280  -1.49   7.54   
4        5  002379      工银香港中小盘人民币  2021-03-16  2.0550  2.0550   2.75   7.54   
...    ...     ...             ...         ...     ...     ...    ...    ...   
7488  7489  011099       富国价值创造混合A  2021-03-17  0.9519  0.9519   0.50          
7489  7490  010953      天弘国证A50指数A  2021-03-17  0.9952  0.9952   0.15          
7490  7491  010107  华夏核心科技6个月定开混合C  2021-03-17  1.0629  1.0629                 
7491  7492  006920       嘉实稳华纯债债券C  2021-03-17  1.0002  1.0002   0.01          
7492  7493  011712       大摩万众创新混合C  2021-03-17  0.8373  0.8373   2.23          

         近1月    近3月    近6月    近1年     近

### 场内交易基金排行榜  
接口: fund_em_exchange_rank  
描述: 获取东方财富网-数据中心-场内交易基金排行榜  
限量: 单次返回当前时刻所有数据, 每个交易日17点后更新  
输入参数  
名称	类型	必选	描述  
-	-	-	-  
输出参数  
名称	类型	默认显示	描述  
序号	int	Y	-  
基金代码	str	Y	-  
基金简称	str	Y	-  
日期	str	Y	-  
单位净值	float	Y	-  
累计净值	float	Y	-  
近1周	float	Y	注意单位: %  
近1月	float	Y	注意单位: %  
近3月	float	Y	注意单位: %  
近6月	float	Y	注意单位: %  
近1年	float	Y	注意单位: %  
近2年	float	Y	注意单位: %  
近3年	float	Y	注意单位: %  
今年来	float	Y	注意单位: %  
成立来	float	Y	注意单位: %  

In [24]:
fund_em_exchange_rank_df = ak.fund_em_exchange_rank()
print(fund_em_exchange_rank_df)

      序号    基金代码           基金简称          日期    单位净值    累计净值   近1周    近1月  \
0      1  515060   华夏中证全指房地产ETF  2021-03-17  0.9842  0.9842  5.48   8.93   
1      2  512200     南方中证房地产ETF  2021-03-17  0.8645  0.8645  5.44   8.50   
2      3  159962    华夏中证四川国改ETF  2021-03-17  1.5542  1.5542  4.91  -8.74   
3      4  159990    银华巨潮小盘价值ETF  2021-03-17  1.1862  1.1862  4.87   7.24   
4      5  512890   华泰柏瑞中证低波动ETF  2021-03-17  1.4275  1.4275  4.86   8.36   
..   ...     ...            ...         ...     ...     ...   ...    ...   
403  404  159880  鹏华国证有色金属行业ETF  2021-03-12  1.0001  1.0001                
404  405  516300  华泰柏瑞中证1000ETF  2021-03-15       1       1                
405  406  159876      华宝有色金属ETF  2021-03-12       1       1                
406  407  159858   南方中证创新药产业ETF  2021-03-12       1       1                
407  408  159871    银华中证有色金属ETF  2021-03-12  0.9995  0.9995                

       近3月    近6月    近1年    近2年     近3年   今年来      成立来  
0    -1.05  -8.95   9.99      

### 货币型基金排行  
接口: fund_em_money_rank  
描述: 获取东方财富网-数据中心-货币型基金排行  
限量: 单次返回当前时刻所有数据, 每个交易日17点后更新, 货币基金的单位净值均为 1.0000 元，最新一年期定存利率: 1.50%  
输入参数  
名称	类型	必选	描述  
-	-	-	-  
输出参数  
名称	类型	默认显示	描述  
序号	int	Y	-  
基金代码	str	Y	-  
基金简称	str	Y	-  
日期	str	Y	-  
万份收益	float	Y	注意单位: %  
年化收益率7日	float	Y	注意单位: %  
年化收益率14日	float	Y	注意单位: %  
年化收益率28日	float	Y	注意单位: %  
近1月	float	Y	注意单位: %  
近3月	float	Y	注意单位: %  
近6月	float	Y	注意单位: %  
近1年	float	Y	注意单位: %  
近2年	float	Y	注意单位: %  
近3年	float	Y	注意单位: %  
近5年	float	Y	注意单位: %  
今年来	float	Y	注意单位: %  
成立来	float	Y	注意单位: %  
手续费	float	Y	-  

In [25]:
fund_em_money_rank_df = ak.fund_em_money_rank()
print(fund_em_money_rank_df)

      序号 基金代码 基金简称     日期       万份收益 年化收益率7日 年化收益率14日 年化收益率28日   近1月    近3月  \
0      1  货币型    1  0.00%     先锋日添利B   0.00%     3.46     6.88  0.67          
1      2  货币型    1  0.00%     先锋日添利A   0.00%     3.77     6.94  0.63          
2      3  货币型    1  0.00%  国泰现金管理货币B   0.00%     3.71     5.85  0.51   12.3   
3      4  货币型    1  0.00%  红土创新优淳货币B   0.00%     5.54     9.61  0.71          
4      5  货币型    1  0.00%     国金众赢货币   0.00%     5.42     9.43  0.69  17.57   
..   ...  ...  ...    ...        ...     ...      ...      ...   ...    ...   
466  467  货币型    1  0.00%     国金鑫盈货币   0.00%     3.98     5.84  0.38  12.49   
467  468  货币型    1  0.00%   兴银现金收益货币   0.00%     4.44     8.23  0.42          
468  469  货币型    1  0.00%    先锋现金宝货币   0.00%     3.22     6.32  0.34          
469  470  货币型    1  0.00%    金信民发货币A   0.00%      3.7     7.19  0.32          
470  471  货币型    1  0.00%  华商现金增利货币A   0.00%     2.99     5.58  0.33  11.96   

      近6月        近1年  近2年     近3年 近5年   今年来      成立

### 理财基金排行  
接口: fund_em_lcx_rank  
描述: 获取东方财富网-数据中心-理财基金排行  
限量: 单次返回当前时刻所有数据, 每个交易日17点后更新, 货币基金的单位净值均为 1.0000 元，最新一年期定存利率: 1.50%  
输入参数  
名称	类型	必选	描述  
-	-	-	-  
输出参数  
名称	类型	默认显示	描述  
序号	int	Y	-  
基金代码	str	Y  	-  
基金简称	str	Y	-  
日期	str	Y	-  
万份收益	float	Y	-  
年化收益率7日	float	Y	注意单位: %  
年化收益率14日	float	Y	注意单位: %  
年化收益率28日	float	Y	注意单位: %  
近1周	float	Y	注意单位: %  
近1月	float	Y	注意单位: %  
近3月	float	Y	注意单位: %  
近6月	float	Y	注意单位: %  
今年来	float	Y	注意单位: %  
成立来	float	Y	注意单位: %  
可购买	float	Y	可购买  
手续费	float	Y	-  

In [26]:
fund_em_lcx_rank_df = ak.fund_em_lcx_rank()
print(fund_em_lcx_rank_df)

None


### 香港基金排行  
接口: fund_em_hk_rank  
描述: 获取东方财富网-数据中心-基金排行-香港基金排行  
限量: 单次返回当前时刻所有数据  
输入参数  
名称	类型	必选	描述  
-	-	-	-  
输出参数  
名称	类型	默认显示	描述  
序号	int	Y	-  
基金代码	str	Y	-  
基金简称	str	Y	-  
币种	str	Y	-  
日期	str	Y	-  
单位净值	float  	Y	-  
日增长率	float	Y	注意单位: %  
近1周	float	Y	注意单位: %  
近1月	float	Y	注意单位: %  
近3月	float	Y	注意单位: %  
近6月	float	Y	注意单位: %  
近1年	float	Y	注意单位: %  
近2年	float	Y	注意单位: %  
近3年	float	Y	注意单位: %  
今年来	float	Y	注意单位: %  
成立来	float	Y	注意单位: %  
可购买	float	Y	1: 可购买 0: 不可购买  
香港基金代码	str	Y	用于查询历史净值数据, 通过该字段查询相关的数据  

In [27]:
fund_em_hk_rank_df = ak.fund_em_hk_rank()
print(fund_em_hk_rank_df)

    序号    基金代码                       基金简称   币种          日期     单位净值    日增长率  \
0    1  968063                  摩根太平洋科技美元   美元  2021-03-17    18.49    0.38   
1    2  968061               摩根太平洋科技人民币对冲  人民币  2021-03-17    18.78  0.3741   
2    3  968062                 摩根太平洋科技人民币  人民币  2021-03-17     17.2  0.4673   
3    4  968041               惠理价值基金P人民币对冲  人民币  2021-03-17  15.7296 -0.1916   
4    5  968066       惠理高息股票基金P每月派息人民币对冲类别  人民币  2021-03-17  14.1427  0.4646   
..  ..     ...                        ...  ...         ...      ...     ...   
83  84  968051              摩根国际债券人民币对冲派息  人民币  2021-03-17    10.09 -0.2964   
84  85  968053                摩根国际债券人民币派息  人民币  2021-03-17     9.78 -0.1021   
85  86  968052                  摩根国际债券人民币  人民币  2021-03-17    10.49 -0.1903   
86  87  968058  博时-安本标准精选新兴市场债券基金PRC类别A美元   美元                                
87  88  968060  博时-安本标准精选新兴市场债券基金PRC类别I美元   美元                                

       近1周      近1月      近3月      近6月      近1年     

## 净值估算  
接口: fund_em_value_estimation  
描述: 获取东方财富网-数据中心-净值估算  
限量: 单次返回当前交易日指定 symbol 的所有数据    
输入参数  
名称	类型	必选	描述  
symbol	str	Y	symbol='全部'; 默认返回所有数据; choice of {'全部', '股票型', '混合型', '债券型', '指数型', 'QDII', 'ETF联接', 'LOF', '场内交易基金'}  
输出参数  
名称	类型	默认显示	描述  
序号	str	Y	-  
基金代码	str	Y	-  
基金名称	str	Y	-  
交易日-估算数据-估算值	float	Y	-  
交易日-估算数据-估算增长率	str	Y	-  
交易日-公布数据-单位净值	float	Y	-  
交易日-公布数据-日增长率	str	Y	-  
估算偏差	str	Y	-  
交易日-单位净值	str	Y	-  

In [28]:
fund_em_value_estimation_df = ak.fund_em_value_estimation(symbol="混合型")
print(fund_em_value_estimation_df)

        序号    基金代码         基金名称 2021-03-18-估算数据-估算值 2021-03-18-估算数据-估算增长率  \
0        1  008293   农银汇理创新医疗混合              1.4891                 3.66%   
1        2  501093   华夏翔阳两年定开混合              1.7529                 3.44%   
2        3  009010   华夏兴阳一年持有混合              1.5836                 3.29%   
3        4  009330    鹏华成长价值混合A              1.2190                 3.29%   
4        5  009331    鹏华成长价值混合C              1.2110                 3.29%   
...    ...     ...          ...                 ...                   ...   
4958  4959  011707    东吴配置优化混合C                 ---                   ---   
4959  4960  010665    博时高端装备混合A                 ---                   ---   
4960  4961  011702  广发睿享稳健增利混合C                 ---                   ---   
4961  4962  011712    大摩万众创新混合C                 ---                   ---   
4962  4963  010699    东方红创新趋势混合                 ---                   ---   

     2021-03-18-公布数据-单位净值 2021-03-18-公布数据-日增长率 估算偏差 2021-03-17-单位净值  
0    

## 基金行情  
接口: fund_etf_hist_sina  
描述: 获取新浪财经-基金行情的日频率行情数据  
限量: 单次返回指定基金的所有数据  
输入参数    
名称	类型	必选	描述  
symbol	str	Y	symbol="sz169103"; 基金列表可以通过 fund_etf_category_sina(symbol="LOF基金") 可选参数为: 封闭式基金, ETF基金, LOF基金 查询  
输出参数  
名称	类型	默认显示	描述  
date	str	Y	-  
open	str	Y	-  
high	float	Y	-  
low	str	Y	-  
close	float	Y	-  
volume	str	Y	注意单位: 手  

In [29]:
fund_etf_hist_sina_df = ak.fund_etf_hist_sina(symbol="sz169103")
print(fund_etf_hist_sina_df)

            date   open   high    low  close  volume
0     2016-08-15  1.069  1.100  1.069  1.100    1200
1     2016-08-16  1.210  1.210  1.080  1.200   15400
2     2016-08-17  1.189  1.199  1.109  1.160   17700
3     2016-08-18  1.117  1.120  1.117  1.120   16100
4     2016-08-19  1.084  1.155  1.084  1.150   18000
...          ...    ...    ...    ...    ...     ...
1086  2021-03-11  1.838  1.838  1.800  1.815  470700
1087  2021-03-12  1.842  1.842  1.812  1.819   65900
1088  2021-03-15  1.800  1.800  1.772  1.772  107400
1089  2021-03-16  1.772  1.800  1.772  1.800  157400
1090  2021-03-17  1.791  1.819  1.791  1.819  230194

[1091 rows x 6 columns]


## 基金持仓  
接口: fund_em_portfolio_hold  
描述: 获取天天基金网-基金档案-投资组合-基金持仓  
限量: 单次返回指定 code 和 year 的所有持仓数据  
输入参数  
名称	类型	必选	描述  
code	str	Y	code="000001"; 基金代码, 可以通过调用 fund_em_fund_name 接口获取  
year	str	Y	year="2020";  
输出参数  
名称	类型	默认显示	描述  
序号	str	Y	-  
股票代码	str	Y	-  
股票名称	str	Y	-  
占净值比例	float	Y	注意单位: %  
持股数	float	Y	注意单位: 万股  
持仓市值	float	Y	注意单位: 万元  
季度	str	Y	-  

In [30]:
fund_em_portfolio_hold_df = ak.fund_em_portfolio_hold(code="000001", year="2020")
print(fund_em_portfolio_hold_df)

     序号    股票代码  股票名称 占净值比例      持股数      持仓市值              季度
0     1  601318  中国平安  6.04   333.75  29029.80  2020年4季度股票投资明细
1     2  002127  南极电商  5.90  2074.47  28378.78  2020年4季度股票投资明细
2     3  000961  中南建设  5.30  2889.73  25516.32  2020年4季度股票投资明细
3     4  600519  贵州茅台  5.01    12.06  24090.89  2020年4季度股票投资明细
4     5  002271  东方雨虹  4.57   566.84  21993.50  2020年4季度股票投资明细
..   ..     ...   ...   ...      ...       ...             ...
152   6  300226  上海钢联  2.75   177.07  12058.31  2020年1季度股票投资明细
153   7  000547  航天发展  2.45   800.00  10760.00  2020年1季度股票投资明细
154   8  300572  安车检测  1.81   204.39   7946.84  2020年1季度股票投资明细
155   9  603259  药明康德  1.65    80.12   7249.88  2020年1季度股票投资明细
156  10  603806   福斯特  1.63   175.98   7165.78  2020年1季度股票投资明细

[157 rows x 7 columns]


## 基金评级  
### 基金评级总汇  
接口: fund_rating_all  
描述: 获取天天基金网-基金评级-基金评级总汇  
限量: 单次返回所有基金评级数据  
输入参数  
名称	类型	必选	描述  
-	-	-	-  
输出参数  
名称	类型	默认显示	描述  
代码	str	Y	-  
简称	str	Y	-  
基金经理	str	Y	-  
基金公司	str	Y	-  
5星评级家数	float	Y	-  
上海证券	float	Y	-  
招商证券	float	Y	-  
济安金信	float	Y	-  
手续费	str	Y	-  
类型	str	Y	-  

In [31]:
fund_rating_all_df = ak.fund_rating_all()
print(fund_rating_all_df)

          代码             简称         基金经理  基金公司 5星评级家数 上海证券 招商证券 济安金信    手续费  \
0     270007       广发大盘成长混合           苗宇    广发      0    4    4    4  0.15%   
1     270009         广发增强债券           谢军    广发      0    2    3    3  0.00%   
2     270010  广发沪深300ETF联接A           刘杰    广发      0         4       0.12%   
3     270014          广发货币B      温秀娟,曾雪兰    广发      0                 0.00%   
4     270025      广发行业领先混合A           程琨    广发      0    5    1    1  0.15%   
...      ...            ...          ...   ...    ...  ...  ...  ...    ...   
5287  001887      中欧价值智选混合E          袁维德    中欧      0              3  0.15%   
5288  004480       华宝智慧产业混合       王正,庄皓亮    华宝      0         3       0.15%   
5289  005427   渤海汇金汇增利3个月定开       李杨,高延龙  渤海汇金      0                 0.08%   
5290  000873       华安现金宝货币A  林唐宇,孙丽娜,李邦长    华安      0                 0.00%   
5291  005378     前海联合泓元定开债券       敬夏玺,张文  前海联合      0                 0.08%   

        类型  
0      混合型  
1      债券型  
2     联接基金  

### 上海证券评级  
接口: fund_rating_sh  
描述: 获取天天基金网-基金评级-上海证券评级  
限量: 单次返回所有基金评级数据  
输入参数  
名称	类型	必选	描述  
-	-	-	-  
输出参数  
名称	类型	默认显示	描述  
代码	str	Y	-  
简称	str	Y	-  
基金经理	str	Y  	-  
基金公司	str	Y	-  
3年期评级-3年评级	float	Y	-  
3年期评级-较上期	float	Y	-  
5年期评级-5年评级	float	Y	-  
5年期评级-较上期	float	Y	-  
单位净值	float	Y	-  
日期	str	Y	-  
日增长率	float	Y	注意单位: %  
近1年涨幅	float	Y	注意单位: %  
近3年涨幅	float	Y	注意单位: %  
近5年涨幅	float	Y	注意单位: %  
手续费	str	Y	-  
类型	str	Y	-  

In [32]:
fund_rating_sh_df = ak.fund_rating_sh()
print(fund_rating_sh_df)

          代码             简称 基金经理  基金公司 3年期评级-3年评级 3年期评级-较上期 5年期评级-5年评级  \
0     160505  博时主题行业混合(LOF)  金晟哲  博时基金          4         2          5   
1     000003       中海可转债债券A  彭海平  中海基金          1         0          1   
2     000004       中海可转债债券C  彭海平  中海基金          1         0          1   
3     000011       华夏大盘精选混合  陈伟彦  华夏基金          5         3          4   
4     000014         华夏聚利债券  何家琪  华夏基金          2         0          2   
...      ...            ...  ...   ...        ...       ...        ...   
2856  750002      安信目标收益债券A  张翼飞  安信基金          5         1          4   
2857  750003      安信目标收益债券C  张翼飞  安信基金          4         0          4   
2858  750005    安信平稳增长混合发起A   张明  安信基金          5         2          5   
2859  762001        国金国鑫发起A   张航  国金基金          3         1          4   
2860  770001         德邦优化混合   汪晖  德邦基金          3         0          2   

     5年期评级-较上期     单位净值          日期  日增长率  近1年涨幅  近3年涨幅   近5年涨幅    手续费   类型  
0            1   1.7980  2021-03-

### 招商证券评级  
接口: fund_rating_zs  
描述: 获取天天基金网-基金评级-招商证券评级  
限量: 单次返回所有基金评级数据  
输入参数    
名称	类型	必选	描述  
-	-	-	-  
输出参数  
名称	类型	默认显示	描述  
代码	str	Y	-  
简称	str	Y	-  
基金经理	str	Y	-  
基金公司	str	Y	-  
3年期评级-3年评级	float	Y	-  
3年期评级-较上期	float	Y	-  
单位净值	float	Y	-  
日期	str	Y	-  
日增长率	float	Y	注意单位: %  
近1年涨幅	float	Y	注意单位: %  
近3年涨幅	float	Y	注意单位: %  
近5年涨幅	float	Y	注意单位: %  
手续费	str	Y	-  

In [33]:
fund_rating_sh_df = ak.fund_rating_sh()
print(fund_rating_sh_df)

          代码             简称 基金经理  基金公司 3年期评级-3年评级 3年期评级-较上期 5年期评级-5年评级  \
0     160505  博时主题行业混合(LOF)  金晟哲  博时基金          4         2          5   
1     000003       中海可转债债券A  彭海平  中海基金          1         0          1   
2     000004       中海可转债债券C  彭海平  中海基金          1         0          1   
3     000011       华夏大盘精选混合  陈伟彦  华夏基金          5         3          4   
4     000014         华夏聚利债券  何家琪  华夏基金          2         0          2   
...      ...            ...  ...   ...        ...       ...        ...   
2856  750002      安信目标收益债券A  张翼飞  安信基金          5         1          4   
2857  750003      安信目标收益债券C  张翼飞  安信基金          4         0          4   
2858  750005    安信平稳增长混合发起A   张明  安信基金          5         2          5   
2859  762001        国金国鑫发起A   张航  国金基金          3         1          4   
2860  770001         德邦优化混合   汪晖  德邦基金          3         0          2   

     5年期评级-较上期     单位净值          日期  日增长率  近1年涨幅  近3年涨幅   近5年涨幅    手续费   类型  
0            1   1.7980  2021-03-

### 济安金信评级  
接口: fund_rating_ja  
描述: 获取天天基金网-基金评级-济安金信评级  
限量: 单次返回所有基金评级数据  
输入参数  
名称	类型	必选	描述  
-	-	-	-  
输出参数  
名称	类型	默认显示	描述  
代码	str	Y	-  
简称	str	Y	-  
基金经理	str	Y	-  
基金公司	str	Y	-  
3年期评级-3年评级	float	Y	-  
3年期评级-较上期	float	Y	-  
单位净值	float	Y	-  
日期	str	Y	-  
日增长率	float	Y	注意单位: %  
近1年涨幅	float	Y	注意单位: %  
近3年涨幅	float	Y	注意单位: %  
近5年涨幅	float	Y	注意单位: %  
手续费	str	Y	-  
类型	str	Y	-  

In [86]:
fund_rating_ja_df = ak.fund_rating_ja()
print(fund_rating_ja_df)

          代码            简称     基金经理    基金公司 3年期评级-3年评级 3年期评级-较上期    单位净值  \
0     000602       富国安益货币A   张波,吴旅忠    富国基金          5         0  0.7498   
1     000009    易方达天天理财货币A  石大怿,刘朝阳   易方达基金          4         0  0.5847   
2     000010    易方达天天理财货币B  石大怿,刘朝阳   易方达基金          4         0  0.6505   
3     000013    易方达天天理财货币R  石大怿,刘朝阳   易方达基金          4         0  0.6532   
4     000198       天弘余额宝货币      王登峰    天弘基金          2         0  0.6198   
...      ...           ...      ...     ...        ...       ...     ...   
4640  660011     农银中证500指数      宋永安  农银汇理基金          3         0  1.5592   
4641  673090    西部利得个股精选股票       刘荟  西部利得基金          3         0  1.7191   
4642  690008  民生中证内地资源主题指数    蔡晓,武杰  民生加银基金          3         0  0.8470   
4643  700002   平安深证300指数增强      毛时超    平安基金          3         0  2.5870   
4644  740101    长安沪深300非周期      林忠晶    长安基金          4         0  1.6370   

              日期    日增长率  近1年涨幅  近3年涨幅  近5年涨幅    手续费   类型  
0     2021-03-17  2.7350   

In [89]:
fund_rating_ja_df['日增长率'] = fund_rating_ja_df['日增长率'].apply(pd.to_numeric, errors='coerce')
fund_rating_ja_df['近1年涨幅'] = fund_rating_ja_df['近1年涨幅'].apply(pd.to_numeric, errors='coerce')
fund_rating_ja_df['近3年涨幅'] = fund_rating_ja_df['近3年涨幅'].apply(pd.to_numeric, errors='coerce')
fund_rating_ja_df['近5年涨幅'] = fund_rating_ja_df['近5年涨幅'].apply(pd.to_numeric, errors='coerce')
fund_rating_ja_df['单位净值'] = fund_rating_ja_df['单位净值'].apply(pd.to_numeric, errors='coerce')
fund_rating_ja_df['3年期评级-较上期'] = fund_rating_ja_df['3年期评级-较上期'].apply(pd.to_numeric, errors='coerce')
fund_rating_ja_df['3年期评级-3年评级'] = fund_rating_ja_df['3年期评级-3年评级'].apply(pd.to_numeric, errors='coerce')

In [90]:
data = fund_rating_ja_df.sort_values(axis=0, ascending=False, by=['日增长率'])
data[0:30]

代码            简称       基金经理     基金公司  3年期评级-3年评级  3年期评级-较上期    单位净值  \
647   690210   民生加银现金增利货币B        李文君   民生加银基金           2        0.0  0.5143   
187   001240   民生加银现金增利货币D        李文君   民生加银基金           2        0.0  0.4663   
646   690010   民生加银现金增利货币A        李文君   民生加银基金           2        0.0  0.4578   
3819  000828        泰达转型机遇         王鹏   泰达宏利基金           5        2.0  1.9890   
2505  002190       农银新能源主题         赵诣   农银汇理基金           4        0.0  2.4513   
3261  162201      泰达宏利成长混合         王鹏   泰达宏利基金           3        0.0  1.4318   
2287  001675         江信同福A     王安良,杨凡     江信基金           1        0.0  1.1560   
2288  001676         江信同福C     王安良,杨凡     江信基金           1        0.0  1.1252   
3374  240004        华宝动力组合        刘自强     华宝基金           3        1.0  2.2182   
467   005151     红土创新优淳货币B         邱骏   红土创新基金           5        0.0  1.2221   
499   091005         大成货币B        陈会荣     大成基金           4        0.0  0.5432   
2175  001261       中融新机遇混合  寇文红,陈荔,汤祺     中融基金           2        0.0  1.1170   
265   002733       上银慧盈利货币        楼昕宇     上银基金           4        1.0  0.7991   
1975  000209      信诚新兴产业混合     王睿,孙浩中   中信保诚基金           3        1.0  2.7610   
3354  213008      宝盈资源优选混合         肖肖     宝盈基金           2        0.0  2.1870   
466   005150     红土创新优淳货币A         邱骏   红土创新基金           5        0.0  1.1566   
383   004179     圆信永丰丰润货币B     林铮,刘莎莎   圆信永丰基金           4        0.0  0.5483   
117   000759       平安财富宝货币         周琛     平安基金           3       -1.0  0.7839   
497   090005         大成货币A        陈会荣     大成基金           4        0.0  0.4838   
4109  003834      华夏能源革新股票        郑泽鸿     华夏基金           5        3.0  2.5020   
203   001669      诺安聚鑫宝货币C    谢志华,周建树     诺安基金           2        0.0  0.7084   
296   003043      交银活期通货币E    连端清,黄莹洁  交银施罗德基金           3       -1.0  1.3022   
2459  002083  新华鑫动力灵活配置混合A         刘彬     新华基金           5        2.0  2.0193   
2460  002084  新华鑫动力灵活配置混合C         刘彬     新华基金           5        2.0  2.0084   
232   001992      农银天天利货币B    史向明,黄晓鹏   农银汇理基金           2        0.0  0.5905   
326   003465      平安金管家货币A        段玮婧     平安基金           3       -1.0  0.8467   
2257  001543       宝盈新锐混合A         肖肖     宝盈基金           5        0.0  2.9670   
3127  020010    国泰金牛创新成长混合      程洲,姜英     国泰基金           3       -1.0  1.4460   
288   002937       华夏沃利货币B     曲波,张雪韬     华夏基金           1       -2.0  0.8481   
3980  001956       国联安科技动力         潘明    国联安基金           3        0.0  1.7972   

              日期   日增长率   近1年涨幅   近3年涨幅   近5年涨幅    手续费   类型  
647   2021-03-17  4.775    2.01    8.46   16.21  0.00%  货币型  
187   2021-03-17  4.592    1.83    7.89   15.20  0.00%  货币型  
646   2021-03-17  4.528    1.77    7.68   14.82  0.00%  货币型  
3819  2021-03-17  4.250   66.30  117.14  165.55  0.12%  股票型  
2505  2021-03-17  3.820  123.64  107.95     NaN  0.15%  混合型  
3261  2021-03-17  3.540  -13.55   32.45   36.30  0.15%  混合型  
2287  2021-03-17  3.510   19.56   10.39   16.27  0.05%  混合型  
2288  2021-03-17  3.510   18.97    8.77   12.97  0.00%  混合型  
3374  2021-03-17  3.510   37.19   58.85   70.16  0.15%  混合型  
467   2021-03-17  3.369    2.52    9.61     NaN  0.00%  货币型  
499   2021-03-17  3.333    2.53    8.57   16.10  0.00%  货币型  
2175  2021-03-17  3.330   18.08   12.60   14.68  0.15%  混合型  
265   2021-03-17  3.279    2.22    8.77     NaN  0.00%  货币型  
1975  2021-03-17  3.250   90.41   94.16   64.74  0.15%  混合型  
3354  2021-03-17  3.220   55.57   51.75   40.09  0.15%  混合型  
466   2021-03-17  3.121    2.28    8.82     NaN  0.00%  货币型  
383   2021-03-17  3.104    2.30    8.73     NaN  0.00%  货币型  
117   2021-03-17  3.098    2.42    9.24   17.43  0.00%  货币型  
497   2021-03-17  3.093    2.29    7.79   14.71  0.00%  货币型  
4109  2021-03-17  3.090  118.32  127.25     NaN  0.15%  股票型  
203   2021-03-17  3.058    2.44    8.69   15.85  0.00%  货币型  
296   2021-03-17  3.054    2.32    8.72     NaN  0.00%  货币型  
2459  2021-03-17  3.040   87.84  112.78     NaN  0.10%  混合型  
2460  2021

## 基金经理  
接口: fund_manager  
描述: 获取天天基金网-基金数据-基金经理大全  
限量: 单次返回所有基金经理数据  
输入参数  
名称	类型	必选	描述  
-	-	-	-  
输出参数  
名称	类型	默认显示	描述  
序号	str	Y	-  
姓名	str	Y	-  
所属公司	str	Y	-  
现任基金	str	Y	-  
累计从业时间	str	Y	注意单位: 天  
现任基金最佳回报	str	Y	注意单位: %  
现任基金资产总规模	str	Y	注意单位: 亿元  

In [35]:
fund_manager_df = ak.fund_manager()
print(fund_manager_df)

100%|██████████| 51/51 [00:24<00:00,  2.06it/s]

      序号   姓名    所属公司                                               现任基金  \
0      1  艾定飞    华商基金                             华商电子行业量化股票,华商计算机行业量化股票   
1      2  艾小军    国泰基金  国泰黄金ETF联接A,国泰黄金ETF联接C,国泰量化成长优选混合A,国泰量化成长优选混合C,...   
2      3   安昀    长信基金            长信内需成长混合E,长信消费升级混合A,长信消费升级混合C,长信内需成长混合A   
3      4   彬彬    人保资产  人保量化混合A,人保量化混合C,人保优势产业混合A,人保优势产业混合C,人保沪深300指数,...   
4      5  包兵华    鹏华基金              鹏华研究驱动混合,鹏华研究智选混合,鹏华成长智选混合A,鹏华成长智选混合C   
...   ..  ...     ...                                                ...   
2520  21  朱征星    富国基金                                           富国金融债债券型   
2521  22  张钟玉    大成基金  大成沪深300指数C,大成MSCI价值100ETF联接A,大成MSCI价值100ETF联接C...   
2522  23  郑振源  创金合信基金  创金合信货币A,创金合信尊享纯债,创金合信尊盛纯债,创金合信尊丰纯债,创金合信尊智纯债,创金...   
2523  24  赵治烨    上银基金  上银新兴价值成长混合,上银鑫达混合,上银未来生活灵活配置混合,上银可转债精选债券,上银内需增...   
2524  25  张子炎    富国基金                      富国鑫旺稳健养老(FOF),富国鑫旺均衡养老三年(FOF)   

     累计从业时间 现任基金最佳回报 现任基金资产总规模  
0       843    52.74      7.23  
1      2621   139.92 

In [36]:
fund_manager_df.to_csv('./fund_manager.csv')

In [37]:
fund_manager_df.columns

Index(['序号', '姓名', '所属公司', '现任基金', '累计从业时间', '现任基金最佳回报', '现任基金资产总规模'], dtype='object')

In [93]:
# data = fund_manager_df['现任基金资产总规模']
# data = data.apply(pd.to_numeric, errors='coerce')
# data
fund_manager_df['现任基金最佳回报'] = fund_manager_df['现任基金最佳回报'].apply(pd.to_numeric, errors='coerce')
data = fund_manager_df.sort_values(axis=0, ascending=False, by=['现任基金最佳回报'])
data[0:20]
# tmp = fund_manager_df['序号']
# tmp = fund_manager_df[fund_manager_df['序号']==37]
# tmp = tmp.sort_values(axis=0, ascending=False, by=['现任基金最佳回报'])
# tmp

序号   姓名      所属公司                                               现任基金  \
2336  37  朱少醒      富国基金                         富国天惠成长混合C,富国天惠成长混合A/B(LOF)   
1891  42  谢治宇    兴证全球基金  兴全合宜混合(LOF)C,兴全社会价值三年持有混合,兴全合润混合(LOF),兴全合宜混合(L...   
1925  26   杨谷      诺安基金                                             诺安先锋混合   
2231  32   张坤     易方达基金        易方达蓝筹精选混合,易方达优质企业三年持有期混合,易方达中小盘混合,易方达亚洲精选股票   
1469  20   魏博      中欧基金  中欧永裕混合A,中欧永裕混合C,中欧盛世成长混合(LOF)E,中欧盛世成长混合(LOF)C,...   
1380  31   孙伟    民生加银基金  民生加银策略精选混合A,民生加银新兴成长混合,民生加银创新成长混合,民生加银新动能一年定开混...   
1820  21   萧楠     易方达基金  易方达瑞恒灵活配置混合,易方达大健康混合,易方达消费精选股票,易方达高质量严选三年持有,易方...   
1307   8   孙芳    上投摩根基金  上投摩根行业睿选股票A,上投摩根行业睿选股票C,上投摩根核心优选混合,上投摩根双息平衡混合A...   
239   40   杜猛    上投摩根基金                上投摩根远见两年持有混合,上投摩根中国优势混合,上投摩根新兴动力混合A   
1924  25   余广    景顺长城基金  景顺长城核心优选混合,景顺长城核心中景一年持有混合,景顺长城核心招景混合,景顺长城核心竞争力混合A   
44    45   陈皓     易方达基金  易方达新经济混合,易方达均衡成长股票,易方达平稳增长混合,易方达科翔混合,易方达科讯混合,易...   
1487  38   王崇   交银施罗德基金                          交银瑞丰三年封闭混合,交银精选混合,交银新成长混合   
2374  25  周蔚文      中欧基金  中欧精选定期开放混合A,中欧新趋势混合E,中欧新蓝筹混合E,中欧精选定期开放混合E,中欧新蓝...   
602    3   姜锋      建信基金       建信健康民生混合,建信环保产业股票,建信优势动力混合(LOF),建信科技创新3年封闭混合   
2392  43  赵晓东  国海富兰克林基金  国富焦点驱动混合,国富恒瑞债券A,国富恒瑞债券C,国富基本面优选混合,国富兴海回报混合,国富...   
879   30   雷鸣     汇添富基金                    汇添富经典成长定开混合,汇添富蓝筹稳健混合,汇添富成长焦点混合   
957    8   李巍      广发基金  广发新兴产业混合A,广发利鑫灵活配置混合A,广发制造业精选混合C,广发新兴产业混合C,广发聚...   
1472  23   魏孛      中金基金  中金绝对收益,中金沪深300A,中金中证500A,中金中证500C,中金沪深300C,中金丰...   
166   17  陈一峰      安信基金  安信价值精选股票,安信消费医药股票,安信新成长混合A,安信新成长混合C,安信价值回报三年持有...   
1949  50   姚锦      建信基金                    建信科技创新3年封闭混合,建信优选成长混合A,建信积极配置混合   

     累计从业时间  现任基金最佳回报 现任基金资产总规模  
2336   5601   2029.53    306.81  
1891   2970    700.82    597.58  
1925   5503    693.47     34.82  
2231   3093    658.15   1197.46  
1469   3137    541.20     20.15  
1380   2446    490.10    233.79  
1820   3093    473.06    626.39  
1307   3388    467.08     99.72  
239    3536    461.80    133.82  
1924   3946    454.39    253.47  
44     3093    421.08    400.96  
1487   2339    412.30    273.12  
2374   5106    411.34    428.32  
602    3537    403.70     25.23  
2392   4214    401.25    142.02  
879    2548    392.24    210.79  
957    3467    391.20     98.28  
1472   1451    382.03     11.04  
166    2523    374.90     93.05  
1949   4096    368.55     25.41

In [ ]:
def init():
    fund_manager_df['现任基金最佳回报'] = fund_manager_df['现任基金最佳回报'].apply(pd.to_numeric, errors='coerce')
    data = fund_manager_df.sort_values(axis=0, ascending=False, by=['现任基金最佳回报'])
    data[0:20]


In [94]:
button = ipywidgets.Button(description="Button")
display(button)
out = ipywidgets.Output()
display(out)
out2 = ipywidgets.Output()
display(out2)

# @out.capture()
def on_button_clicked(b):
    global count
    print(f'[-] hello button.')
    with out:
        print(f'[=] hello button.')
        show()
    with out2:
        out2.clear_output()
        plot([count+2, count+3], [count+4, count+5])
        show()
    count += 1
        
button.on_click(on_button_clicked)

NameError: name 'ipywidgets' is not defined

In [103]:
import pandas as pd
import numpy as np
import random

In [104]:
a = np.array(np.linspace(0, 11, 12)).reshape(3, 4)
a

array([[ 0.,  1.,  2.,  3.],
       [ 4.,  5.,  6.,  7.],
       [ 8.,  9., 10., 11.]])

In [105]:
pda = pd.DataFrame(data=a, columns=['a', 'b', 'c', 'd'])
pda

a    b     c     d
0  0.0  1.0   2.0   3.0
1  4.0  5.0   6.0   7.0
2  8.0  9.0  10.0  11.0

In [110]:
data = [1, 2, 3]
target = [4, 5, 6]

In [109]:
tmp = zip(data, target)
random.shuffle(tmp)
data[:], target[:] = zip(*tmp)

TypeError: object of type 'zip' has no len()

In [111]:
zip2list = [i for i in zip(data, target)]
data[:], target[:] = zip(*random.shuffle(zip2list))

TypeError: type object argument after * must be an iterable, not NoneType

In [112]:
aa = [1, 2, 3, 4, 5, 6]
bb = [1, 2, 3, 4, 5, 6]

cc = list(zip(aa, bb))
random.shuffle(cc)
aa[:], bb[:] = zip(*cc)
print(aa, bb)

[4, 6, 1, 2, 3, 5] [4, 6, 1, 2, 3, 5]


In [113]:
aa = np.array([[1., 0.], [2., 1.], [0., 0.]])
bb = np.array([0, 1, 2])
cc = list(zip(aa, bb))
random.shuffle(cc)
aa[:], bb[:] = zip(*cc)
print(aa, bb)

[[2. 1.]
 [2. 1.]
 [0. 0.]] [1 0 2]


In [114]:
import random

In [115]:
random.randint(0, 0)

0